// philosophy/node/node_modules/product.js
let priceSchema = require('./priceSchema.js')
let functions = require('unconscious')
let defaultParser = functions['defaultParser.js']
let schema = {
	"path" : "thing",
	"properties": [
		{
			path: 'title',
			mongoose: {
				schema: {
					default: 'unknown',
					type: String,
					required: true
				}
			}
		},
		{
			path: 'names',
			mongoose: {
				schema: {
					default: 'unknown',
					type: [String]
				}
			}
		},
		{
			path: 'stock',
			parser: defaultParser,
			mongoose: {
				schema: {
					default: 'unknown',
					type: String,
					required: true
				}
			}
		},
		{
			path: 'type',
			mongoose: {
				schema: {
					default: 'unknown',
					type: String,
				}
			}
		},
		{
			path: 'subtitle',
			mongoose: {
				schema: {
					default: 'unknown',
					type: String,
				}
			}
		},
		{
			path: 'short description',
			mongoose: {
				schema: {
					default: 'unknown',
					type: String,
				}
			}
		},
		{
			path: 'prices',
			mongoose: {
				schema: {
					default: [
						priceSchema.mongoose.default
					],
					type: [
						priceSchema.mongoose.schema
					],
					required: true
				}
			}
		},
		{
			path: 'thumbnail',
			selector: ".product-image > img",
			attribute: 'src',
			mongoose: {
				schema: {
					default: 'unknown',
					type: String,
					required: true
				}
			}
		},
		{
			path: 'images',
			mongoose: {
				schema: {
					default: ['unknown'],
					type: [String]
				}
			}
		},
		{
			path: 'source',
			page: true,
			selector: "#header > div > div.row > div.col-md-10.col-xs-12.col-sm-12 > div.col-md-6.col-xs-12.col-sm-12 > a > img.large",
			attribute: 'alt',
			mongoose: {
				schema: {
					default: 'unknown',
					type: String,
				}
			}
		}
	]
}
module.exports = schema
