// philosophy/node/node_modules/quickBloomLightsProductSchema.js
let productSchema = require('./product.js')
let functions = require('experiences')
let defaultParser = functions['defaultParser.js']

function priceParser(args){
	var ret = undefined
	if(args.val && typeof args.val == 'string'){
		args.val = args.val.replace(/[\$,]/ig, '')
		// let currency = args.val.slice(0, 2)+'D'
		ret = {
			name: args.property.name,
			values: {
				'AUD': parseFloat(args.val)
			},
			source: 'quick bloom lights'
		}
	}
	return ret
}

module.exports = {
	"path" : "thing",
	schema: productSchema,
	"properties": [
		{
			path: 'title',
			html: {
				selector: "div.sub > a",
				attribute: 'innerHTML',
			}
		},
		{
			path: 'names',
			html: {
			}
		},
		{
			path: 'stock',
			html: {
				selector: "div.main > a > span.soldout > span",
				attribute: 'innerText',
			}
		},
		{
			path: 'type',
			html: {
				page: true,
				selector: "#shopify-section-collection-template > div > div.content-header.align-centre > h1",
			}
		},
		{
			path: 'subtitle',
			html: {
				selector: "",
				attribute: '',
			}
		},
		{
			path: 'short description',
			html: {
				selector: "",
				attribute: 'innerHTML',
			}
		},
		{
			path: 'prices',
			html: {
				properties: [
					{
						name: 'quick bloom lights old price',
						parser: priceParser,
						html: {
							selector: "div.sub > div > span.was-price",
							attribute: 'innerHTML',
						}
					},
					{
						name: 'quick bloom lights regular price',
						parser: priceParser,
						html: {
							selector: "div.sub > div > span.price",
							attribute: 'innerHTML',
						}
					},
					{
						name: 'quick bloom lights sale price',
						parser: priceParser,
						html: {
							selector: "div.sub > div > span.price",
							attribute: 'innerHTML',
						}
					},
				],
			}
		},
		{
			path: 'thumbnail',
			html: {
				selector: "div.main > a > img",
				attribute: 'src',
			}
		},
		{
			path: 'images',
			html: {
			}
		},
		{
			path: 'source',
			html: {
				page: true,
				selector: "#navpanel > div.shoplogo > a > span > img",
				attribute: 'alt',
			}
		}
	],
	methods: {
		pricesai: functions['productpricesai.js']
	},
	ai: functions['productai.js']
}
